message/messaging_service.hh

```diff
@@ -64,14 +65,6 @@ class frozen_mutation;
 class frozen_schema;
 class canonical_mutation;
 
-namespace dht {
-    class token;
-    class ring_position;
-    using partition_range = interval<ring_position>;
-    using token_range = interval<token>;
-    using token_range_vector = std::vector<token_range>;
-}
-
```

Comment:
Hmm, I'm not sure why it was done in this series.
Since it's the last patch in the series, indeed nothing depends on it at the moment.

Reply:
I'll send it separately since in general I don't like forward declaration that include types (like `using token_range_vector = std::vector<token_range>`) and not just class names, to be distributed all over the place.
We better have one source of truth for them in the include _fwd header file.

Reply:
I agree (and the patch is good) but this is a lost cause.

To get a single source of truth, we need C++ modules.

Reply:
dropped this patch in v4